File: apps/job-admin/src/app/core/types/Dtos/CreateJobRequest.ts

export type JobType = 'fullTime' | 'partTime' | 'contract' | 'internship' | 'temporary' | 'other';

export const JOB_TYPE_LABELS: Record<JobType, string> = {
  fullTime: 'Full Time',
  partTime: 'Part Time',
  contract: 'Contract',
  internship: 'Internship',
  temporary: 'Temporary',
  other: 'Other',
};

export interface CreateJobDto {
  title: string;
  companyUId: string;
  location: string;
  aboutRole: string;
  jobType: JobType;
  salaryRange?: string | null;
  responsibilities: string[];
  qualifications: string[];
  draftId?: string;
  deleteDraft: boolean;
}

File: services/monolith/Src/Core/JobBoard.Contracts/Jobs/CreateJobRequest.cs
namespace JobBoard.Monolith.Contracts.Jobs;

public class CreateJobRequest 
{
    public required string Title { get; set; }
    public Guid CompanyUId { get; set; }
    public required string Location { get; set; }
    public required JobType JobType { get; set; }
    public required string AboutRole { get; set; }
    public string? SalaryRange { get; set; }
    public List<string> Responsibilities { get; set; } = [];
    public List<string> Qualifications { get; set; } = [];
    public string DraftId {get;set;} = null!;
    public bool DeleteDraft {get;set;} = false;
}

public enum JobType
{
    FullTime,
    PartTime,
    Contract,
    Internship
}